src/dolphin/downloadDolphin.ts

```diff
@@ -27,106 +39,63 @@ export async function assertDolphinInstallation(
-
 async function compareDolphinVersion(type: DolphinLaunchType, latestVersion: string): Promise<boolean> {
   const dolphinPath = await findDolphinExecutable(type);
   const dolphinVersion = spawnSync(dolphinPath, ["--version"]).stdout.toString();
   return lt(dolphinVersion, latestVersion);
 }
 
-export async function openDolphin(type: DolphinLaunchType, params?: string[]): Promise<ChildProcessWithoutNullStreams> {
-  const dolphinPath = await findDolphinExecutable(type);
-  return spawn(dolphinPath, params);
-}
-
-async function getLatestDolphinAsset(type: DolphinLaunchType): Promise<any> {
-  const release = await getLatestReleaseData(type);
-  const asset = release.assets.find((a: any) => matchesPlatform(a.name));
-  if (!asset) {
-    throw new Error(`No release asset matched the current platform: ${process.platform}`);
-  }
-  return asset;
-}
-
-async function getLatestReleaseData(type: DolphinLaunchType): Promise<any> {
-  const owner = "project-slippi";
-  let repo = "Ishiiruka";
-  if (type === DolphinLaunchType.PLAYBACK) {
-    repo += "-Playback";
-  }
-  return getLatestRelease(owner, repo);
-}
-
-function matchesPlatform(releaseName: string): boolean {
-  switch (process.platform) {
-    case "win32":
-      return releaseName.endsWith("Win.zip");
-    case "darwin":
-      return releaseName.endsWith("Mac.dmg");
-    case "linux":
-      return releaseName.endsWith("Linux.zip");
-    default:
-      return false;
-  }
+export async function downloadAndInstallDolphin(
+  type: DolphinLaunchType,
+  releaseInfo: DolphinVersionResponse,
+  log: (message: string) => void,
+  cleanInstall = false,
+): Promise<void> {
+  const downloadedAsset = await downloadLatestDolphin(releaseInfo, log);
+  log(`Installing v${releaseInfo.version} ${type} Dolphin...`);
+  await installDolphin(type, downloadedAsset, log, cleanInstall);
+  log(`Finished v${releaseInfo.version} ${type} Dolphin install`);
 }
 
 async function downloadLatestDolphin(
-  type: DolphinLaunchType,
+  releaseInfo: DolphinVersionResponse,
   log: (status: string) => void = console.log,
 ): Promise<string> {
-  const asset = await getLatestDolphinAsset(type);
+  const downloadUrl = releaseInfo.downloadUrls[process.platform];
+  const parsedUrl = new URL(downloadUrl);
+  const filename = path.basename(parsedUrl.pathname);
+
   const downloadDir = path.join(app.getPath("userData"), "temp");
   await fs.ensureDir(downloadDir);
-  const downloadLocation = path.join(downloadDir, asset.name);
+  const downloadLocation = path.join(downloadDir, filename);
   const exists = await fileExists(downloadLocation);
   if (!exists) {
-    log(`Downloading ${asset.browser_download_url} to ${downloadLocation}`);
+    log(`Downloading ${downloadUrl} to ${downloadLocation}`);
     const win = BrowserWindow.getFocusedWindow();
     if (win) {
-      await download(win, asset.browser_download_url, {
-        filename: asset.name,
+      await download(win, downloadUrl, {
+        filename: filename,
```

Comment:
This is just a comment, but I think it would be good if we could eventually not depend on the existence of a browser window to download a file since `electron-dl` seems to require a Browser window. I feel like there are a few cases where the app is running without a window (e.g. MacOS) or in the task bar/system tray (perhaps in the future).